common/step_provision.go

```diff
@@ -39,13 +47,26 @@ func PopulateProvisionHookData(state multistep.StateBag) map[string]interface{}
 	id, ok := state.GetOk("instance_id")
 	if ok {
 		hookData["ID"] = id
-	} else {
-		// Warn user that the id isn't implemented
-		hookData["ID"] = "ERR_ID_NOT_IMPLEMENTED_BY_BUILDER"
 	}
 
 	hookData["PackerRunUUID"] = os.Getenv("PACKER_RUN_UUID")
-	hookData["PackerHTTPAddr"] = GetHTTPAddr()
+
+	// Packer HTTP info
+	hookData["PackerHTTPIP"] = HttpIPNotImplemented
+	hookData["PackerHTTPPort"] = HttpPortNotImplemented
+	hookData["PackerHTTPAddr"] = HttpAddrNotImplemented
+
+	httpPort, okPort := state.GetOk("http_port")
+	if okPort {
+		hookData["PackerHTTPPort"] = strconv.Itoa(httpPort.(int))
```

Comment:
Another thing that confused me is that in the `shell-local` provisioner `PACKER_HTTP_PORT` is used as a string whereas here we use `PackerHTTPPort` is used as an int. This totally confused me for a minute, but this has nothing to do with this PR.